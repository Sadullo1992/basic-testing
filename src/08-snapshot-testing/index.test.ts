import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const tree = generateLinkedList([
      '<a href="https://nodejs.org">Nodejs</a>',
      '<a href="http://www.twitter.com">Twitter</a>',
    ]);
    expect(tree).toStrictEqual({
      next: {
        next: {
          next: null,
          value: null,
        },
        value: '<a href="http://www.twitter.com">Twitter</a>',
      },
      value: '<a href="https://nodejs.org">Nodejs</a>',
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const tree = generateLinkedList([
      '<a href="http://www.facebook.com">Facebook</a>',
      '<a href="http://www.instagram.com">Instagram</a>',
    ]);
    expect(tree).toMatchSnapshot();
  });
});
