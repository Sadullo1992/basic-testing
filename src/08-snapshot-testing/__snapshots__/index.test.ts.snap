// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": null,
      "value": null,
    },
    "value": "<a href="http://www.instagram.com">Instagram</a>",
  },
  "value": "<a href="http://www.facebook.com">Facebook</a>",
}
`;
